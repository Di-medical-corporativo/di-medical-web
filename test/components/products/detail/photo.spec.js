import Photo from '@/components/products/detail/photo.vue';
import { shallowMount } from '@vue/test-utils';

describe('photo detail component', () => {

    test('should match with snapshot', () => {
        const wrapper = shallowMount(Photo);
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('main image should has first images of images', () => {
        const wrapper = shallowMount(Photo);
        expect(wrapper.vm.mainImage.image).toBe(wrapper.vm.images[0]);
    });

    test('should change image with setMainImage method', () => {
        const wrapper = shallowMount(Photo);
        wrapper.vm.setMainImage('https://rukminim2.flixcart.com/image/714/857/jcrz6vk0/shoe/g/w/t/ds-1503-10-d-sneakerz-white-original-imafftryhbku8nus.jpeg?q=50', 2);
        expect(wrapper.vm.mainImage).toEqual({
            image: 'https://rukminim2.flixcart.com/image/714/857/jcrz6vk0/shoe/g/w/t/ds-1503-10-d-sneakerz-white-original-imafftryhbku8nus.jpeg?q=50',
            idx: 2
        })
    });

    test('should call setMainImage when click at thumnail', async () => {
        const wrapper = shallowMount(Photo);
        const changeImageFunc = (wrapper.vm.setMainImage = jest.fn());
        await wrapper.findAll('div.product__detail__photo__thumbnail').at(1).trigger('click');
        expect(changeImageFunc).toHaveBeenCalledWith("https://flexi.shoes/blog/wp-content/uploads/sites/7/2018/06/sneaker-clasico-blanco.png", 1)
    });

});