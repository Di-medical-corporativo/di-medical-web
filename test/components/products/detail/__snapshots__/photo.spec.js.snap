// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photo detail component should match with snapshot 1`] = `
"<div class=\\"product__detail__photo\\">
  <div class=\\"product__detail__photo__main\\" style=\\"background-image: url(https://media.istockphoto.com/photos/pair-of-stylish-sneakers-isolated-on-white-background-white-casual-picture-id1248566148?k=20&amp;m=1248566148&amp;s=170667a&amp;w=0&amp;h=Lom1xUV8pfz2kFmzepcEzn52mlEzWj5d2TMDFKlhOsc=);\\">
    <div class=\\"prodoct__detail__full_size\\">
      <b-icon icon=\\"arrows-fullscreen\\" class=\\"fullscreen__icon\\"></b-icon>
    </div>
  </div>
  <div class=\\"product__detail__photo__list\\">
    <div id=\\"0\\" class=\\"product__detail__photo__thumbnail current_thumbnail\\" style=\\"background-image: url(https://media.istockphoto.com/photos/pair-of-stylish-sneakers-isolated-on-white-background-white-casual-picture-id1248566148?k=20&amp;m=1248566148&amp;s=170667a&amp;w=0&amp;h=Lom1xUV8pfz2kFmzepcEzn52mlEzWj5d2TMDFKlhOsc=);\\"></div>
    <div id=\\"1\\" class=\\"product__detail__photo__thumbnail\\" style=\\"background-image: url(https://flexi.shoes/blog/wp-content/uploads/sites/7/2018/06/sneaker-clasico-blanco.png);\\"></div>
    <div id=\\"2\\" class=\\"product__detail__photo__thumbnail\\" style=\\"background-image: url(https://rukminim2.flixcart.com/image/714/857/jcrz6vk0/shoe/g/w/t/ds-1503-10-d-sneakerz-white-original-imafftryhbku8nus.jpeg?q=50);\\"></div>
    <div id=\\"3\\" class=\\"product__detail__photo__thumbnail\\" style=\\"background-image: url(https://media.istockphoto.com/photos/one-white-sneaker-and-floating-rope-isolated-on-white-background-with-picture-id1040446784?k=20&amp;m=1040446784&amp;s=612x612&amp;w=0&amp;h=SM0ccxM_Xwua8xFc0YsB4gCTjm4_KpDeE1ppd8TewBM=);\\"></div>
  </div>
</div>"
`;
